Rubberduck.Inspections/Concrete/UnreachableCaseInspectionValue.cs

```diff
@@ -0,0 +1,619 @@
+            }
+            else if (textValue.Equals(Tokens.True) || textValue.Equals(Tokens.False))
+            {
+                return Tokens.Boolean;
+            }
+            else if (long.TryParse(textValue, out _))
+            {
+                return Tokens.Long;
+            }
+            else
+            {
+                return defaultType;
+            }
+        }
+
+        public string UseageTypeName
+        {
+            set
+            {
+                if (value != _useageTypeName)
+                {
+                     _useageTypeName = value;
+                    _operatorIsGT = GetDelegate(OperatorsIsGT, _useageTypeName);
+                    _operatorIsLT = GetDelegate(OperatorsIsLT, _useageTypeName);
+                    _operatorIsEQ = GetDelegate(OperatorsIsEQ, _useageTypeName);
+                    _opMult = GetDelegate(OperatorsMult, _useageTypeName);
+                    _opDiv = GetDelegate(OperatorsDiv, _useageTypeName);
+                    _opPlus = GetDelegate(OperatorsPlus, _useageTypeName);
+                    _opMinus = GetDelegate(OperatorsMinus, _useageTypeName);
+                }
+            }
+            get { return _useageTypeName; }
+        }
+
+        public string DerivedTypeName
+        {
+            get
+            {
+                if (_derivedTypeName.Equals(string.Empty))
+                {
+                    _derivedTypeName = DeriveTypeName(_inputString, UseageTypeName);
+                }
+                return _derivedTypeName;
+            }
+        }
+
+        public bool IsStringConstant => _ctorValueIsString;
+
+        public bool HasValue
+            => HasValueTests.ContainsKey(UseageTypeName) ? HasValueTests[UseageTypeName](this) : false;
+
+        public bool IsWithin(UnreachableCaseInspectionValue start, UnreachableCaseInspectionValue end, bool isInclusive = true ) 
+            => isInclusive ?
+                start > end ? this >= end && this <= start : this >= start && this <= end
+                : start > end ? this > end && this < start : this > start && this < end;
+
+        public bool ExceedsMaxMin()
+              => MaxMinTests.ContainsKey(UseageTypeName) ? MaxMinTests[UseageTypeName](this) : false;
+
+        public static bool operator >(UnreachableCaseInspectionValue thisValue, UnreachableCaseInspectionValue compValue)
+        {
+            return thisValue._operatorIsGT(thisValue, compValue);
+        }
+
+        public static bool operator <(UnreachableCaseInspectionValue thisValue, UnreachableCaseInspectionValue compValue)
+        {
+            return thisValue._operatorIsLT(thisValue, compValue);
+        }
+
+        public static bool operator ==(UnreachableCaseInspectionValue thisValue, UnreachableCaseInspectionValue compValue)
+        {
+            if(ReferenceEquals(null, thisValue))
+            {
+                return ReferenceEquals(null, compValue);
+            }
+            else
```

Comment:
opening the else block here is not necessary :)

Reply:
Cleaned-up (and others like it)